### PR TITLE
Added Flatpak deployment section

### PR DESCRIPTION
**Pull request by AxelSilverdew**

### CODE REVIEW

File: deployment/flatpak/about.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
++title: Flatpak
++subsection: flatpak
++section: deployment
++description: >
++Flatpak (formerly xdg-app) is a software utility for software deployment, package management and application virutalization. It provides a sandbox environment in which users can run applications in isolation from the rest of the system.
++---
++
++# {{page.title}}
++
++[Flatpak](https://flatpak.org/) is a software utility for software deployment, package management and application virtualization. It provides a sandbox environment in which users can run applications in isolation from the rest of the system.
```
Comment (pvalena) [inline deployment/flatpak/about.md]: Please create something like `References` sub-header (double hashtag - `## References`) with a link to documentation and you should provide at least basic usage, including installation (is there something you need to install to use Flatpacks on Fedora?) and something about where-to-find-them etc..

Description is not nearly enough I'd say.

**Comment by pvalena** [comment]
You do not need to create another pull reqest to change the commit / edit your PR changes. Etc.
Instead, the usual workflow is to add commits and/or amend them (`git commit -a --amend` to add all changes to your existing commit).

**Comment by pvalena** [comment]
(and simply push the changes)

File: deployment/flatpak/about.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+title: Flatpak
+subsection: flatpak
+section: deployment
+description: >
+	Flatpak (formerly xdg-app) is a software utility for software deployment, package 	management and application virutalization. It provides a sandbox environment in which users 	can run applications in isolation from the rest of the system.
```
Comment (pvalena) [inline deployment/flatpak/about.md]: This is still not fixed. (See https://github.com/developer-portal/content/pull/247#discussion_r163558675)

File: deployment/flatpak/usage.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
+title: Flatpak Usage
+subsection: flatpak
+section: deployment
```
Comment (pvalena) [inline deployment/flatpak/usage.md]: You need to add order to all pages. Example:
https://github.com/developer-portal/content/blame/master/tech/languages/haskell/haskell-installation.md#L5

File: deployment/flatpak/about.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+[Flatpak](https://flatpak.org/) is a software utility for software deployment, package management and application virtualization. It provides a sandbox environment in which users can run applications in isolation from the rest of the system.
+
+
+
+## Advantages of Using Flatpak
+
+* Flatpaks are distro agnostic. Developers can distribute their application on every Linux distribution simultaneously with ease.
+* Flatpaks provide consistent environments. Developers can develop and test applications in environments that are identical to the one users have.
+* Flatpak's build tools are simple and very easy to use. A link to the full documentation can be found in the "Further Reading" section below.
```
Comment (pvalena) [inline deployment/flatpak/about.md]: >  A link to the full documentation can be found in the "Further Reading" section below.

please remove this sentence

File: deployment/flatpak/about.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+* Flatpak runtimes provide stable sets of dependencies that can be used by applications.
```
Comment (pvalena) [inline deployment/flatpak/about.md]: Can be quite a disadvantage. I'd rather leave that out.

File: deployment/flatpak/about.md (hunk 2 above)
Comment (AxelSilverdew) [reply deployment/flatpak/about.md]: I'm sorry, can you please explain what I'm supposed to fix?
I don't think I'm following. 😅

File: deployment/flatpak/about.md (hunk 4 above)
Comment (AxelSilverdew) [reply deployment/flatpak/about.md]: Okay, noted.

File: deployment/flatpak/about.md (hunk 5 above)
Comment (AxelSilverdew) [reply deployment/flatpak/about.md]: Leave out the advantages section?

File: deployment/flatpak/about.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+* Dependencies that aren't in a runtime can be bundled as part of an application. Versions of the dependencies that are used can be fixed and patched.
```
Comment (pvalena) [inline deployment/flatpak/about.md]: This seems to go against principles that are customary to Fedora. Is this also the case on Fedora?
Keep in mind this should be Fedora specific. General comparisons can be linked to.

File: deployment/flatpak/about.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+* Flatpaks are future-proof. Runtimes are forward compatible, irrespective of distribution changes. Flatpak applications run on new Linux distribution versions as they are released.
```
Comment (pvalena) [inline deployment/flatpak/about.md]: This highly depends on the content itself, right? Do ve have some initiative on that in Fedora? In upstream? Can this be linked to?
I cannot think of a value that this bring to a Fedora user/developer. This is a 'selling point' (vision?) and not an real advantage IMHO.

File: deployment/flatpak/about.md [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+
+
+## Further Reading
+
+* [Flatpak Installation](/deployment/flatpak/installation.html)
```
Comment (pvalena) [inline deployment/flatpak/about.md]: No need for this sort of links. This is another page and thus showed in the sub-menu on left.

**Comment by pvalena** [comment]
What is you source for the content? It must me distributable, or free etc. Otherwise we'll have to take it back.

File: deployment/flatpak/about.md (hunk 2 above)
Comment (pvalena) [reply deployment/flatpak/about.md]: The explanation is linked in the comment.
Please join our #developer-portal irc chanell on [Freenode](https://freenode.net/) so I can explain details there and give you some additional advice.

File: deployment/flatpak/about.md (hunk 5 above)
Comment (pvalena) [reply deployment/flatpak/about.md]: I meant just the line I've commented on.

**Comment by pvalena** [comment]
Come to think of it, if you take this as a user tooling, it should probably be in `tools` section.

Altogether, its philosophy seems to be very similar to docker, so the pages could be similar too IMO:
https://developer.fedoraproject.org/tools/docker/about.html

(wchich can be of a help to you)

File: deployment/flatpak/about.md (hunk 2 above)
Comment (AxelSilverdew) [reply deployment/flatpak/about.md]: Okay, joining.

**Comment by RheaAyase** [comment]
Yeah I like to compare it as "docker for gui desktop applications..." while docker is for serverside stuff.

File: deployment/flatpak/install.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+section: deployment
+---
+
+# {{page.title}}
+
+## Installing Flatpak
+
+A `flatpak` package has been available for Fedora 24+ and has been installed by default on Fedora Workstation.
+
+If you do not use Fedora Workstation, you can install Flatpak by running:
```
Comment (pvalena) [inline deployment/flatpak/install.md]: I'd rather say 'if you're not sure, run this command to install....'

**Comment by AxelSilverdew** [comment]
> What is you source for the content? It must me distributable, or free etc. Otherwise we'll have to take it back.
All the content comes from the official Flatpak Documentation, and the Fedora Magazine article on Flatpak, both of which have been linked within the pages.

> Its philosophy seems to be very similar to docker
Yeah, they're very similar. As @RheaAyase mentioned, Flatpak can be thought of as "Docker for Desktop Applications."

File: deployment/flatpak/install.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+```
+$ sudo dnf install flatpak
+```
+
+
+
+## Installing Flatpak Builder
+
+Most applications require additional dependencies that aren’t provided by their runtimes. Flatpak allows these dependencies to be bundled as part of the application itself. In order to do this, each dependency
```
Comment (pvalena) [inline deployment/flatpak/install.md]: Casual user has to do this, or only if you want to create flatpaks yourself? If the latter one, please split this to a standalone page.

Please try to focus on best practices for Developers **on** Fedora.
See our [wiki](https://fedoraproject.org/wiki/Websites/Developer) for more info.

Comment (AxelSilverdew) [reply deployment/flatpak/install.md]: Flatpak Builder is for developers who are interested in packaging their applications into flatpaks. Not for the end users.

I'll split it into another page as per your suggestion.

**Comment by AxelSilverdew** [comment]
> Come to think of it, if you take this as a user tooling, it should probably be in tools section.

The primary use of Flatpak _is_ easy packaging and deployment though. Even the [Fedorawiki](https://fedoraproject.org/wiki/Flatpak) defines flatpak as "a cross-distribution packaging format".

Personally, I think it belongs in `deployment`, but I'll move it into tools if you feel like that's more appropriate.

File: deployment/flatpak/install.md (hunk 10 above)
Comment (pvalena) [reply deployment/flatpak/install.md]: `# Building Flatpacks`

_Note to self: we should probably point to that page from deployment/distribution section._

File: deployment/flatpak/install.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+```
+$ sudo dnf install flatpak-builder
+```
+
+
+
+## Further Reading
+
+* [FedoraMagazine - Getting Started with Flatpak](https://fedoramagazine.org/getting-started-flatpak/)
+* [Get Flatpak](https://flatpak.org/getting.html)
```
Comment (pvalena) [inline deployment/flatpak/install.md]: No need to point from everypage. And most importantly- do content duplication please...

File: deployment/flatpak/usage.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+# {{page.title}}
+
+## Installing Flatpak Applications
+
+### Setting up the Flathub repository:
```
Comment (pvalena) [inline deployment/flatpak/usage.md]: I'd say these are too many nestings for something as simple ... sooo, skip the `## Installing Flatpak Applications` ?

File: deployment/flatpak/install.md (hunk 11 above)
Comment (AxelSilverdew) [reply deployment/flatpak/install.md]: Content duplication? What do you mean? 😅

File: deployment/flatpak/usage.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+section: deployment
+---
+
+# {{page.title}}
+
+## Installing Flatpak Applications
+
+### Setting up the Flathub repository: 
+
+A team of Flatpak developers have started a project known as [Flathub](https://flathub.org). Flathub aims to provide a centralized repository for making Flatpak applications available to users. Flathub covers more than the GNOME application suite, and is regularly adding new applications.
```
Comment (pvalena) [inline deployment/flatpak/usage.md]: This is copied from external source, right? I'm not sure this is the right way- maybe just point to it and in one your own sentence explain like 'you can get the comminuty flatpacks from....`.

File: deployment/flatpak/install.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+# {{page.title}}
+
+## Installing Flatpak
+
+A `flatpak` package has been available for Fedora 24+ and has been installed by default on Fedora Workstation.
+
+If you do not use Fedora Workstation, you can install Flatpak by running:
+
+```
+$ sudo dnf install flatpak
```
Comment (pvalena) [inline deployment/flatpak/install.md]: This should probable be on the first page `about = usage`.

File: deployment/flatpak/usage.md (hunk 13 above)
Comment (AxelSilverdew) [reply deployment/flatpak/usage.md]: This was taken from the FedoraMagazine article, yes.

Alright, I'll make the necessary changes.

File: deployment/flatpak/usage.md [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+
+To add the Flathub repository, open the Terminal, and run:
+
+```
+$ flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
```
Comment (pvalena) [inline deployment/flatpak/usage.md]: Is some local directory created? I think you should elaborate what it does and what to do with the result etc.

File: deployment/flatpak/usage.md [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+```
+
+
+
+### Installing Applications (GNOME Software): 
+
+GNOME Software already supports Flatpak repositories, so applications can be installed directly from GNOME Software.
```
Comment (pvalena) [inline deployment/flatpak/usage.md]: And user runs f.e. KDE?
What then?

Comment (AxelSilverdew) [reply deployment/flatpak/usage.md]: Non-GNOME users are recommended to use the Command Line instructions.
I should probably have mentioned that somewhere.
I'll add a sentence about it.

File: deployment/flatpak/about.md (hunk 6 above)
Comment (AxelSilverdew) [reply deployment/flatpak/about.md]: > This seems to go against principles that are customary to Fedora.

Can you elaborate please?

Comment (pvalena) [reply deployment/flatpak/about.md]: Yes, on IRC.

**Comment by pvalena** [comment]
> All the content comes from the official Flatpak Documentation, and the Fedora Magazine article on Flatpak, both of which have been linked within the pages.

I was a bit more worried about 'legalities' of things. The content of a page is often copyrighted, while we do release it as community content, thus redistributable. Therefore it _may_ be better just to link to that content than to copy paste it (within reason - short descriptions, etc. are fine.)

File: deployment/flatpak/usage.md (hunk 16 above)
Comment (pvalena) [reply deployment/flatpak/usage.md]: Great!

File: deployment/flatpak/usage.md (hunk 13 above)
Comment (pvalena) [reply deployment/flatpak/usage.md]: No need to copy whole blocks of text I'd say. See my other comment.

File: deployment/flatpak/install.md (hunk 11 above)
Comment (pvalena) [reply deployment/flatpak/install.md]: I mean - there are the same links on every page... It's fine to have them just on `Page 1`, or even better, with the corresponding text.

(Just as an example: when you mention `copr`, you ought link the text, and/or put a reference to some copr guide or something).

**Comment by pvalena** [comment]
> Yeah, they're very similar. As @RheaAyase mentioned, Flatpak can be thought of as "Docker for Desktop Applications."

Yes, I know that (nothing new here for me).
I'm just trying to hint to:
   (1) you can use that- it's a nice way to recognise what it does for a developer who already knows docker (is there some comparison page?) and
   (2) you can look at the docker subsection and maybe get some idea what can the content for Flatpak section look like.

**Comment by AxelSilverdew** [comment]
> you can look at the docker subsection and maybe get some idea what can the content for Flatpak section look like.

Yeah, I'm making edits based on the docker page now.
Plus a little clean up here and there.

**Comment by AxelSilverdew** [comment]
@pvalena sorry this took so long, I got caught up in work. Can you please let me know if there's any more changes I need to make?

**Comment by pvalena** [comment]
Np, I've got a lot of other work too, I'll look into this later on.

File: deployment/flatpak/flatpak-building.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
...
+
+If you've decided to package and distribute your application as a flatpak, then [this tutorial](https://flatpak.org/hello-world.html) is a good place to start learning how to create a flatpak.
+
+It'll walk you through installing the flatpak sdk and packaging your very first application.
+
+
+
+## References
+
+* [Official Flatpak Documentation](http://docs.flatpak.org/en/latest/)
```
Comment (pvalena) [inline deployment/flatpak/flatpak-building.md]: References here do not make sense IMHO, I'll delete it myself.

**Comment by pvalena** [comment]
LGTM.

**Pull request merged**